sklearn/tests/test_docstring_parameters_consistency.py

```diff
@@ -1,13 +1,64 @@
 # Authors: The scikit-learn developers
 # SPDX-License-Identifier: BSD-3-Clause
 
+from typing import Any
+
 import pytest
 
 from sklearn import metrics
-from sklearn.ensemble import StackingClassifier, StackingRegressor
+from sklearn.ensemble import (
+    AdaBoostClassifier,
+    AdaBoostRegressor,
+    StackingClassifier,
+    StackingRegressor,
+)
+from sklearn.mixture import BayesianGaussianMixture, GaussianMixture
 from sklearn.utils._testing import assert_docstring_consistency, skip_if_no_numpydoc
 
-CLASS_DOCSTRING_CONSISTENCY_CASES = [
+CLASS_DOCSTRING_CONSISTENCY_CASES: list[dict[str, Any]] = [
+    {
+        "objects": [AdaBoostClassifier, AdaBoostRegressor],
+        "include_params": True,
+        "exclude_params": None,
+        "include_attrs": True,
+        "exclude_attrs": [
+            # type differs
+            "estimators_",
+            "",
+        ],
+        "include_returns": False,
+        "exclude_returns": None,
+        "descr_regex_patterns": {
+            # Excludes 2nd sentence if present, matches last sentence until "."
+            "estimator": r"^([^.]+\.)(?:\s+[^.]+\.)?\s+([^.]+\.)",
```

Comment:
I'm worried that we are introducing too much complexity with those regex.

I'm wondering if we could come with a small utility to express what we want to achieve without writing a regex (but I don't have a solution).

Reply:
Completely agree, the regexes turned out to be way more complicated than initially thought. 

I don't have a great solution either. What about a `descr_ignore` that takes a string that is removed from the description? This way you enter the sentence/word that is different, and all occurrences will be removed from the description before matching.
We can use `re.sub` and replace with empty string.

Seems like it would be easier to use but is it a bit weird that you are entering the portion that is to be removed ?

cc @adrinjalali

Reply:
@glemaitre suggested that we could simplify this by vectorize the descriptions and calculate the cosine similarity and allow users to simply specific specific parameters where, 'close enough' is acceptable?

WDYT @adrinjalali  ?

Reply:
O_o, that's an interesting idea. But I'm afraid in the context of docstrings, contradictory docstrings can have very close cosine similarities, so that's probably mostly unhelpful.

I'm probably missing / forgetting something, but could we start w/o regex and see how much of our docstrings we can test this way? Like exact match or "includes" kinda thing. My thinking is that we don't have to support every single scenario in our codebase. If we can support most docstrings, we're fine?

Reply:
Yes I think I agree with you.

> I'm probably missing / forgetting something, but could we start w/o regex and see how much of our docstrings we can test this way?

This is a good point. I do think the current `descr_regex_pattern` is not a great solution - I don't like how we have to re-write the whole docstring and the regex pattern is difficult to implement. 

Why don't I remove `descr_regex_pattern` and we add some tests, only for objects that are most pertinent. In the process of adding tests, we may even think of a decent solution...?

The question though, is do we create a meta issue and open it up to everyone or....?

cc @glemaitre

Reply:
We can open a meta issue, but I personally feel like similar past ones (me being guilty of creating some of them) have distracted us quite a bit from other high priority projects in our roadmap. So I probably would be happy if we end up with a few large-ish PRs that more experienced people like us do?